### DISCUSSION TITLE
Play video in original FPS, and just 5 FPS send to inference and draw masks  on result

### DISCUSSION THREAD

**Discussion by damavand1** [ultralytics/ultralytics#8449] [General]
Video file or camera is 60 FPS, I want play 60 FPS in output to show result very smooth, but i want send just 5 FPS for inference to YOLO8 model and draw masks on 60 FPS result

I'm sure 5 FPS for inference is enough for me, and my low power computer can calculate it

this is my code 

`import cv2
from ultralytics import YOLO
import random

model = YOLO("D:\\Pirooz\\0 handwrite detector\\runs\\segment\\train\\weights\\best.pt")  # load a pretrained model
video_path ="D:\\Pirooz testing\\20240222_103752.mp4"

cap=cv2.VideoCapture(video_path)

original_width = int(cap.get(cv2.CAP_PROP_FRAME_WIDTH))
original_height = int(cap.get(cv2.CAP_PROP_FRAME_HEIGHT))

target_width = 640
target_height = int(original_height * target_width / original_width)

total_frames = int(cap.get(cv2.CAP_PROP_FRAME_COUNT))
start_frame = random.randint(0, total_frames - 1)

cap.set(cv2.CAP_PROP_POS_FRAMES, start_frame)

while cap.isOpened():
    success, frame = cap.read()

    if success:
        frame = cv2.resize(frame, (target_width, target_height))

        results = model(frame)

        annotated_frame = results[0].plot()

        cv2.imshow("Annotated Frame", annotated_frame)

        cv2.imshow("Original Frame", frame)

        if cv2.waitKey(1) & 0xFF == ord("q"):
            break 

    else:
        break

cap.release()
cv2.destroyAllWindows()

`

how can i do this ?

tnx

**Comment by glenn-jocher** [comment]
Hello! 😊 To achieve smooth playback at 60 FPS while sending only 5 FPS for inference to the YOLOv8 model, you can modify your loop to only perform inference on every 12th frame (assuming your video is 60 FPS, this equates to 5 FPS for inference). Here's a simplified approach:

```python
import cv2
from ultralytics import YOLO

model = YOLO("path/to/your/model.pt")  # Load your pretrained model
video_path = "path/to/your/video.mp4"

cap = cv2.VideoCapture(video_path)
frame_id = 0  # Initialize frame counter

while cap.isOpened():
    success, frame = cap.read()
    if not success:
        break  # Exit loop if no more frames

    # Perform inference only on every 12th frame
    if frame_id % 12 == 0:
        results = model(frame)
        annotated_frame = results[0].plot()
        cv2.imshow("Annotated Frame", annotated_frame)
    else:
        cv2.imshow("Original Frame", frame)  # Show original frame

    if cv2.waitKey(1) & 0xFF == ord("q"):
        break

    frame_id += 1  # Increment frame counter

cap.release()
cv2.destroyAllWindows()
```

This code ensures that your video plays back smoothly by displaying every frame but only sends a subset of frames for inference, thus reducing the computational load on your computer. Let me know if this helps or if you have any more questions!

**Reply by damavand1** [reply]
Hi
thank you for you support

now, the problem is video plays at least 4x fasting than normal play
it's because while loop repeat very fast and faster than 60 fps on my computer

this technique is good when i reading from camera but in file video plays very fast

**Reply by glenn-jocher** [reply]
@damavand1 hi there! 😊

Thanks for reaching out, and I'm glad to hear you're finding the support helpful!

For the issue of the video playing too fast, it's indeed because the loop processes frames quicker than real-time playback. To match the video's original FPS, you can add a delay based on the video's FPS. Here's a quick example:

```python
import cv2
from ultralytics import YOLO

model = YOLO("path/to/your/model.pt")
video_path = "path/to/your/video.mp4"

cap = cv2.VideoCapture(video_path)
fps = cap.get(cv2.CAP_PROP_FPS)  # Get the original FPS of the video
delay = int(1000 / fps)  # Calculate delay in milliseconds between frames

while cap.isOpened():
    success, frame = cap.read()
    if not success:
        break

    # Your inference code here

    if cv2.waitKey(delay) & 0xFF == ord("q"):  # Use the calculated delay
        break

cap.release()
cv2.destroyAllWindows()
```

This should help synchronize the playback speed with the original video FPS. Let me know if this solves your issue or if you have any more questions!
